web/src/components/wizard/SetupStep.tsx

```diff
@@ -142,38 +164,69 @@ const SetupStep: React.FC<SetupStepProps> = ({ onNext }) => {
-          <p className="text-gray-600 mt-1">Configure the installation settings.</p>
+          <p className="text-gray-600 mt-1">
+            {currentView === 'configuration' 
+              ? "Configure the installation settings." 
+              : "Validate the host requirements before proceeding with installation."
+            }
+          </p>
         </div>
 
-        {isLoading ? (
-          <div className="py-4 text-center">
-            <div className="animate-spin rounded-full h-8 w-8 border-b-2 border-gray-900 mx-auto"></div>
-            <p className="mt-2 text-gray-600">Loading configuration...</p>
-          </div>
+        {currentView === 'configuration' ? (
+          // Configuration View
+          <>
+            {isLoading ? (
+              <div className="py-4 text-center">
+                <div className="animate-spin rounded-full h-8 w-8 border-b-2 border-gray-900 mx-auto"></div>
+                <p className="mt-2 text-gray-600">Loading configuration...</p>
+              </div>
+            ) : (
+              <LinuxSetup
+                config={config}
+                prototypeSettings={prototypeSettings}
+                showAdvanced={showAdvanced}
+                onShowAdvancedChange={setShowAdvanced}
+                onInputChange={handleInputChange}
+                onSelectChange={handleSelectChange}
+                availableNetworkInterfaces={availableNetworkInterfaces}
+                fieldErrors={submitError?.errors || []}
+              />
+            )}
+
+            {error && (
+              <div className="mt-4 p-3 bg-red-50 text-red-500 rounded-md">
+                Please fix the errors in the form above before proceeding.
+              </div>
+            )}
+          </>
         ) : (
-          <LinuxSetup
-            config={config}
-            prototypeSettings={prototypeSettings}
-            showAdvanced={showAdvanced}
-            onShowAdvancedChange={setShowAdvanced}
-            onInputChange={handleInputChange}
-            onSelectChange={handleSelectChange}
-            availableNetworkInterfaces={availableNetworkInterfaces}
-            fieldErrors={submitError?.errors || []}
-          />
-        )}
-
-        {error && (
-          <div className="mt-4 p-3 bg-red-50 text-red-500 rounded-md">
-            Please fix the errors in the form above before proceeding.
-          </div>
+          // Validation View
+          <LinuxPreflightCheck onComplete={handlePreflightComplete} />
         )}
       </Card>
 
-      <div className="flex justify-end">
-        <Button onClick={handleNext} icon={<ChevronRight className="w-5 h-5" />}>
-          Next: Validate Host
-        </Button>
+      <div className="flex justify-between">
+        {currentView === 'validation' && (
+          <Button variant="outline" onClick={handleBackToConfiguration} icon={<ChevronLeft className="w-5 h-5" />}>
+            Back to Configuration
```

Comment:
I prefer just "Back" on this button. Configuration is going to be confusing anyway once we add the actual config screen into the installer.